api/tests/test_views.py

```diff
@@ -1,179 +1,177 @@
+            },
+            {
+              'question': 'question2 description',
+              'reading': 'question2 reading',
+              'answers': [
+                  {
+                    'answer': 'answer1 description',
+                    'correct': False
+                  },
+                  {
+                    'answer': 'answer2 description',
+                    'correct': True
+                  },
+                  {
+                    'answer': 'answer3 description',
+                    'correct': False
+                  },
+                  {
+                    'answer': 'answer4 description',
+                    'correct': False
+                  }
+                ]
+              }   
+          ]
+     }
+ }
+
+    response = self.client.post('/api/v1/teachers/%s/lessons/' % self.teacher1.id, data=data, content_type='application/json')
+
+    self.assertEqual(response.status_code, 201)
+    self.assertEqual(response.data['name'], data['lesson']['name'])
+    self.assertIsInstance(response.data['questions'][0], dict)
+    self.assertIsInstance(response.data['questions'][1], dict)
+    self.assertIsInstance(response.data['questions'][0]['answers'][0], dict)
+    self.assertIsInstance(response.data['questions'][1]['answers'][0], dict)
+
+  def test_can_get_lesson(self):
+    response = self.client.get('/api/v1/lessons/%s/' % self.lesson1.id)
+
+    self.assertEqual(response.status_code, 200)
+    self.assertEqual(response.data['name'], self.lesson1.name)
+    self.assertIsInstance(response.data['questions'][0], dict)
+    self.assertIsInstance(response.data['questions'][1], dict)
+    self.assertIsInstance(response.data['questions'][0]['answers'][0], dict)
+    self.assertIsInstance(response.data['questions'][1]['answers'][0], dict)
+
+
+class StudentLessonViewSet(TestCase):
+  def setUp(self):
+    self.teacher = Teacher.objects.create(first_name='Severus', last_name='Snape')
+    self.student = Student.objects.create(first_name='Draco', last_name='Malfoy', teacher_id=self.teacher.id)
+    self.lesson = self.teacher.lesson_set.create(name='Potions', description='Brew potions properly')
+
+  def test_post_student_lesson(self):
+    data = {
+      "lesson": self.lesson.id,
+      "score": 3,
+      "mood": "I had a brilliant time"
+    }
+    response = self.client.post('/api/v1/students/%s/' % (self.student.id), data=data, content_type='application/json')
    
-#     self.assertEqual(response.status_code, 200)
-#     self.assertEqual(response.data['score'], 3)
-#     self.assertEqual(response.data['mood'], 'I had a brilliant time')
-#     self.assertEqual(response.data['lesson'], 4)
-#     self.assertEqual(response.data['student'], 3)
+    self.assertEqual(response.status_code, 200)
+    self.assertEqual(response.data['score'], 3)
+    self.assertEqual(response.data['mood'], 'I had a brilliant time')
+    self.assertEqual(response.data['lesson'], 4)
+    self.assertEqual(response.data['student'], 3)
```

Comment:
Hopefully we don't have to keep commenting and uncommenting tests out with any new migrations